pkgs/applications/window-managers/wayfire/default.nix

```diff
@@ -1,33 +1,89 @@
-{ lib, stdenv, fetchurl, cmake, meson, ninja, pkg-config
-, cairo, doctest, libdrm, libexecinfo, libinput, libjpeg, libxkbcommon, wayland
-, wayland-protocols, wf-config, wlroots, mesa
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, meson
+, ninja
+, pkg-config
+, cairo
+, doctest
+, libdrm
+, libexecinfo
+, libinput
+, libjpeg
+, libxkbcommon
+, wayland
+, wayland-protocols
+, wayland-scanner
+, wlroots
+, mesa
+, pango
+, xorg
+, libevdev
+, libxml2
+, glm
+, useSystemWlroots ? true
 }:
 
 stdenv.mkDerivation rec {
   pname = "wayfire";
-  version = "0.7.2";
+  version = "0.7.5";
 
-  src = fetchurl {
-    url = "https://github.com/WayfireWM/wayfire/releases/download/v${version}/wayfire-${version}.tar.xz";
-    sha256 = "1gasijjyfl00zpy6j9hh6qpwv0sw42h9irycbnm693j3vw9mcy66";
+  src = fetchFromGitHub {
+    owner = "WayfireWM";
+    repo = pname;
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-Z+rR9pY244I3i/++XZ4ROIkq3vtzMgcxxHvJNxFD9is=";
   };
 
-  nativeBuildInputs = [ cmake meson ninja pkg-config wayland ];
-  buildInputs = [
-    cairo doctest libdrm libexecinfo libinput libjpeg libxkbcommon wayland
-    wayland-protocols wf-config wlroots mesa
+  nativeBuildInputs = [
+    cmake
+    meson
+    ninja
+    pkg-config
```

Comment:
Usually only 3 of them are required. I subject cmake could be removed, but not sure.

Reply:
meson use cmake to find `doctest`,   I tend to put them in `nativeCheckInputs`